prov/tcp/src/tcpx_ep.c

```diff
@@ -41,6 +41,82 @@
 extern struct fi_ops_rma tcpx_rma_ops;
 extern struct fi_ops_msg tcpx_msg_ops;
 
+void tcpx_hdr_ntoh(struct tcpx_base_hdr *hdr)
+{
+	struct ofi_rma_iov *rma_iov;
+	uint8_t *ptr = (uint8_t *)hdr + sizeof(*hdr);
+	int i;
+
+	hdr->flags = ntohs(hdr->flags);
+	hdr->size = ntohll(hdr->size);
+
+	if (hdr->flags & OFI_REMOTE_CQ_DATA) {
+		*((uint64_t *)ptr) = ntohll(*((uint64_t *) ptr));
+		ptr += sizeof(uint64_t);
+	}
+
+	rma_iov = (struct ofi_rma_iov *)ptr;
+	for ( i = 0; i < hdr->rma_iov_cnt; i++) {
+		rma_iov[i].addr = ntohll(rma_iov[i].addr);
+		rma_iov[i].len = ntohll(rma_iov[i].len);
+		rma_iov[i].key = ntohll(rma_iov[i].key);
+	}
+}
+
+static void tcpx_hdr_hton(struct tcpx_base_hdr *hdr)
+{
+	struct ofi_rma_iov *rma_iov;
+	uint8_t *ptr = (uint8_t *)hdr + sizeof(*hdr);
+	int i;
+
+
+	if (hdr->flags &OFI_REMOTE_CQ_DATA) {
+		*((uint64_t *)ptr) = htonll(*((uint64_t *) ptr));
+		ptr  += sizeof(uint64_t);
+	}
+
+	rma_iov = (struct ofi_rma_iov *)ptr;
+	for ( i = 0; i < hdr->rma_iov_cnt; i++) {
+		rma_iov[i].addr = htonll(rma_iov[i].addr);
+		rma_iov[i].len = htonll(rma_iov[i].len);
+		rma_iov[i].key = htonll(rma_iov[i].key);
+	}
+
+	hdr->flags = htons(hdr->flags);
+	hdr->size = htonll(hdr->size);
+}
```

Comment:
You can merge the above functions into a single function, since hton == ntoh.  I.e. you can define tcpx_hdr_hton as tcpx_hdr_ntoh.  Or rename to tcpx_hdr_swap.

Reply:
I've come to realize that this whole patch can be simplified. I will rework.